extras/corelens.py

```diff
@@ -0,0 +1,140 @@
+                return
+
+            vmcore_dir_path = Path(vmcore_dir_path)
+            if not vmcore_dir_path.is_dir():
+                error_logs.append(f"Error: {vmcore_dir_path} is not a valid directory.")
+                self.add_string_as_file("\n".join(error_logs), filename="corelens")
+                return
+
+            now = datetime.now()
+            oldest_allowed_date = now - timedelta(days)
+            recent_vmcore_dirs = []
+
+            for directory in vmcore_dir_path.iterdir():
+                if directory.is_dir():
+                    try:
+                        timestamp_str = directory.name[-19:]
+                        dir_timestamp = datetime.strptime(timestamp_str, "%Y-%m-%d-%H:%M:%S")
+                        if dir_timestamp >= oldest_allowed_date:
+                            recent_vmcore_dirs.append((dir_timestamp, directory))
+                    except ValueError:
+                        error_logs.append(f"Warning: Skipping invalid directory format: {directory.name}")
+
+            recent_vmcore_dirs.sort(reverse=True, key=lambda x: x[0])
+            recent_vmcore_dirs = [directory[1] for directory in recent_vmcore_dirs[:3]]
+            if not recent_vmcore_dirs:
+                error_logs.append("No recent vmcore directories found.")
+                self.add_string_as_file("\n".join(error_logs), filename="corelens")
+                return
+
+            archive_tmp_dir = self.archive.get_tmp_dir()
+            corelens_output_path = Path(archive_tmp_dir) / "sos_commands/corelens"
+            corelens_output_path.mkdir(parents=True, exist_ok=True)
+
+            kernel_version = subprocess.check_output("uname -r", shell=True, universal_newlines=True).strip()
+            vmlinux_ctf_path = f"/usr/lib/modules/{kernel_version}/kernel/vmlinux.ctfa"
+
+            for vmcore_subdir in recent_vmcore_dirs:
+                vmcore_file = vmcore_subdir / "vmcore"
+                corelens_output_file = corelens_output_path / vmcore_subdir.name
+                corelens_cmd = f"corelens --ctf {vmlinux_ctf_path} {vmcore_file} -A -o {corelens_output_file}"
+                try:
+                    self.add_cmd_output(corelens_cmd, suggest_filename=f"corelens-{vmcore_subdir.name}")
+                except subprocess.CalledProcessError:
+                    error_logs.append(f"Error: Failed to process vmcore at {vmcore_file}")
+        except Exception as e:
+            error_logs.append(f"Unexpected error: {e}")
+
+        if error_logs:
+            self.add_string_as_file("\n".join(error_logs), filename="corelens")
+
+    def setup(self):
+        days = self.get_option('task-days')
+        corelens_binary = Path("/usr/bin/corelens")
+        ctf_path = Path(f"/usr/lib/modules/{subprocess.check_output('uname -r', shell=True).decode('utf-8').strip()}/kernel/vmlinux.ctfa")
+        if corelens_binary.exists() and ctf_path.exists() and self.check_version():

```

Comment:
Again, I'm not certain why this is here. There's no reason to assume that the vmcore is from the same kernel version as the current kernel. Let corelens determine that.